File: 0002-add-two-numbers/0002-add-two-numbers.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *dummy=new ListNode();
        ListNode *current=dummy;
        int c=0;
        while(l1 || l2 ||c){
            int x=l1 ? l1->val  : 0;
            int y=l2 ?l2->val: 0;
            int sum=c+x+y;
            c=floor(sum/10);
            current->next=new ListNode(sum%10);
            current=current->next;
            l1=l1? l1->next:NULL;
            l2=l2? l2->next:NULL;

        } 
        return dummy->next;
    }
};

File: 0002-add-two-numbers/0002-add-two-numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def addTwoNumbers(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        str1=""
        str2=""
        while l1:
            str1+=str(l1.val)
            l1=l1.next
        while l2:
            str2+=str(l2.val)
            l2=l2.next
        ans=int(str1[::-1])+int(str2[::-1])
        l=[]
        for i in str(ans):
            l.append(i)
        l=l[::-1]
        dummy=ListNode(0)
        cop=dummy
        for (i) in l:
            cop.next=ListNode(int(i))
            cop=cop.next
        return dummy.next